### PR TITLE
Add forms, comments, security and change DB to Postgres :)

### PR DESCRIPTION
**Pull request by DrEdi**

### CODE REVIEW

**Comment by webknjaz** [comment]
Name the PR properly

File: blog/settings.py [hunk 1]
```diff
@@ -104,9 +109,22 @@
...
+DATABASES['default'] = dj_database_url.config()
+
+SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDED_PROTO', 'https')
+
+ALLOWED_HOSTS = ['*']
+
+DEBUG = False
+
+try:
+    from .local_settings import BASE_DIR, DATABASES, DEBUG  # noqa: F401
```
Comment (webknjaz) [inline blog/settings.py:119]: You don't need the `noqa` comment anymore

File: main/models.py [hunk 2]
```diff
@@ -23,3 +23,26 @@ def __str__(self):
...
+    created_date = models.DateTimeField(default=timezone.now)
+    approved_comment = models.BooleanField(default=False)
+
+    def approve(self):
+        """Approve comment and save it in DB."""
+        self.approved_comment = True
+        self.save()
+
+    def __str__(self):
+        """Return comment text."""
```
Comment (webknjaz) [inline main/models.py]: I see that it returns comment's text field value. It's stated right on the next line. Write something useful here instead. Why is this method needed at all?

Comment (webknjaz) [reply main/models.py]: I would write `Render Comment instance as it's text by default when stringifying`

File: main/templates/main/post_draft_list.html [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{% extends 'main/base.html' %}
+
+{% block content %}
+    {% for post in posts %}
+        <div class="post">
+            <p class="date">created: {{ post.created_date|date:'d-m-Y' }}</p>
+            <h1><a href="{% url 'post_detail' pk=post.pk %}">{{ post.title }}</a></h1>
+            <p>{{ post.text|truncatechars:200 }}</p>
```
Comment (webknjaz) [inline main/templates/main/post_draft_list.html]: It is common to wrap pipe with spaces.

File: main/views.py [hunk 4]
```diff
@@ -13,3 +15,88 @@ def post_detail(request, pk):
...
+@login_required
+def post_remove(request, pk):
+    """View for deleting posts."""
+    post = get_object_or_404(Post, pk=pk)
+    post.delete()
+    return redirect('post_list')
+
+
+def add_comment_to_post(request, pk):
+    """If someone want to create new comment he/she get this view."""
```
Comment (webknjaz) [inline main/views.py]: In English combo `he/she` is often referred to as `them`.
P.S. `want` -> `wants`

File: main/views.py [hunk 5]
```diff
@@ -13,3 +15,88 @@ def post_detail(request, pk):
...
+
+
+@login_required
+def post_edit(request, pk):
+    """View is for editing post."""
+    post = get_object_or_404(Post, pk=pk)
+    if request.method == 'POST':
+        form = PostForm(request.POST, instance=post)
+        if form.is_valid():
+            post = form.save(commit=False)
```
Comment (webknjaz) [inline main/views.py:41]: Nit: in future I would encapsulate this into a method.

**Comment by DrEdi** [comment]
Done :+1:

**Review by webknjaz — CHANGES_REQUESTED** [review]
Please make your PR look awesome

File: blog/settings.py (hunk 1 above)
Comment (DrEdi) [reply blog/settings.py:119]: There is one problem without `noqa`:
`blog/settings.py:128:5: F401 '.local_settings.DEBUG' imported but unused`

**Comment by webknjaz** [comment]
What are the extensions

File: main/models.py [hunk 6]
```diff
@@ -44,5 +44,5 @@ def approve(self):
         self.save()
 
     def __str__(self):
-        """Return comment text."""
+        """Render Comment instance as it's text by default when stringifying."""
```
Comment (webknjaz) [inline main/models.py]: `it's` -> `its`

File: blog/settings.py [hunk 7]
```diff
@@ -66,8 +67,12 @@
 
 DATABASES = {
     'default': {
-        'ENGINE': 'django.db.backends.sqlite3',
-        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
```
Comment (webknjaz) [inline blog/settings.py:70]: I would move this to `local_settings` instead of wiping completely

File: blog/settings.py [hunk 8]
```diff
@@ -66,8 +67,12 @@
...
+        'ENGINE': 'django.db.backends.postgresql_psycopg2',
```
Comment (webknjaz) [inline blog/settings.py]: Why do you even need this? You completely replace `DATABASES['default']` below

File: blog/settings.py (hunk 1 above)
Comment (webknjaz) [reply blog/settings.py:119]: ok

File: main/forms.py [hunk 9]
```diff
@@ -0,0 +1,19 @@
+"""This module is for python forms."""
```
Comment (webknjaz) [inline main/forms.py]: Oh really? What is this supposed to mean?

File: main/models.py [hunk 10]
```diff
@@ -23,3 +23,26 @@ def __str__(self):
...
+    created_date = models.DateTimeField(default=timezone.now)
+    approved_comment = models.BooleanField(default=False)
+
+    def approve(self):
+        """Approve comment and save it in DB."""
+        self.approved_comment = True
+        self.save()
+
+    def __str__(self):
+        """Render Comment instance as it's text by default when stringifying."""
```
Comment (webknjaz) [inline main/models.py]: `it's` -> `its`

**Review by webknjaz — CHANGES_REQUESTED** [review]

File: blog/settings.py (hunk 7 above)
Comment (DrEdi) [reply blog/settings.py:70]: This is there
But `local_settings` are ignored by .gitignore file

File: main/forms.py [hunk 11]
```diff
@@ -0,0 +1,23 @@
+"""This module contains classes that represent Forms.
+
+It's easier to use built-in forms because you don't need to think about
+connection with DB, SQL query etc. Just write FormModel and views for it
+"""
```
Comment (webknjaz) [inline main/forms.py]: Оце виглядає як пропаганда джанго, а не опис призначення модуля.
Ось як я б це записав: `This module keeps the collection of form declarations representing HTML forms binded to data in the DB`.
На відміну від твого варіанту, тут є чітко окреслене призначення модуля і опис його вмісту, без реклами джанго і смс :)

File: main/models.py [hunk 12]
```diff
@@ -23,3 +23,26 @@ def __str__(self):
         This is useful if you need to get info only about title of Post object
```
Comment (webknjaz) [inline main/models.py]: This is a lie! If I need a title of post I'll use `Post.title` explicitly. See Zen of Python (PEP 20): **Explicit is better than implicit**.

You should've stated that this is used when rendering model instance as a string.

File: blog/settings.py [hunk 13]
```diff
@@ -104,9 +109,22 @@
...
 
-
+LOGIN_REDIRECT_URL = '/'
 # Static files (CSS, JavaScript, Images)
 # https://docs.djangoproject.com/en/1.10/howto/static-files/
 
 STATIC_URL = '/static/'
 STATIC_ROOT = os.path.join(BASE_DIR, 'static')
+
+DATABASES['default'] = dj_database_url.config()
```
Comment (webknjaz) [inline blog/settings.py]: It's worth declaring right in that dict above.

**Review by webknjaz — CHANGES_REQUESTED** [review]

File: blog/settings.py [hunk 14]
```diff
@@ -104,9 +109,24 @@
...
 
-
+LOGIN_REDIRECT_URL = '/'
 # Static files (CSS, JavaScript, Images)
 # https://docs.djangoproject.com/en/1.10/howto/static-files/
 
 STATIC_URL = '/static/'
 STATIC_ROOT = os.path.join(BASE_DIR, 'static')
+
+DATABASES = {
```
Comment (webknjaz) [inline blog/settings.py]: Why did you paste it here instead of replacing the one above?

**Review by webknjaz — CHANGES_REQUESTED** [review]

File: requirements.txt [hunk 15]
```diff
@@ -0,0 +1,12 @@
+aspy.yaml==0.2.2
```
Comment (webknjaz) [inline requirements.txt]: Are all of these dependencies really needed?

Comment (webknjaz) [reply requirements.txt]: Could you please split ones needed always from those needed in production?

Comment (webknjaz) [reply requirements.txt]: You may check the way it works in @pavlenk0's [repo](https://github.com/kpi-web-guild/django-girls-blog-pavlenk0/blob/master/test_requirements.txt)

**Review by webknjaz — APPROVED** [review]

File: requirements.txt (hunk 15 above)
Comment (DrEdi) [reply requirements.txt]: I'm not sure,
I did it from virt environment, so I thought all from there are important

Comment (webknjaz) [reply requirements.txt]: Some of them are dependencies of others and do not need to be declared explicitly. It's a mess.

Comment (DrEdi) [reply requirements.txt]: One moment, I'll fix it!

File: blog/settings.py [hunk 16]
```diff
@@ -104,9 +102,20 @@
 
 USE_TZ = True
 
-
+LOGIN_REDIRECT_URL = '/'
```
Comment (webknjaz) [inline blog/settings.py:105]: I would add a new line next to this one

File: main/models.py [hunk 17]
```diff
@@ -20,6 +20,29 @@ def publish(self):
...
+
+
+class Comment(models.Model):
+    """Model for comments."""
+
+    post = models.ForeignKey('main.Post', related_name='comments')
+    author = models.CharField(max_length=200)
+    text = models.TextField()
+    created_date = models.DateTimeField(default=timezone.now)
+    approved_comment = models.BooleanField(default=False)
```
Comment (webknjaz) [inline main/models.py:39]: Nit: it would be better to call this column `is_approved`

File: main/models.py [hunk 18]
```diff
@@ -20,6 +20,29 @@ def publish(self):
...
         """Return title of Post.
 
-        This is useful if you need to get info only about title of Post object
+        This is used when rendering model instance as a string
         """
         return self.title
+
+    def approved_comments(self):
+        """Approving comments. Use for moderating."""
+        return self.comments.filter(approved_comment=True)
```
Comment (webknjaz) [inline main/models.py:29]: I would add a method to the `Comment` model to encapsulate filtering in there.

File: main/models.py [hunk 19]
```diff
@@ -20,6 +20,29 @@ def publish(self):
     def __str__(self):
         """Return title of Post.
 
-        This is useful if you need to get info only about title of Post object
+        This is used when rendering model instance as a string
         """
         return self.title
+
+    def approved_comments(self):
+        """Approving comments. Use for moderating."""
```
Comment (webknjaz) [inline main/models.py:28]: `Approving` is Present Continuous Tense. Write it in imperative mode. Also you didn't describe an action this method does.

File: main/models.py (hunk 17 above)
Comment (DrEdi) [reply main/models.py:39]: Oooops, sorry, I thought I can merge it...
I'll fix it next time

**Pull request merged**